File: ProblemSolving/Warmup/time_conversion.py
import os
import sys


def timeConversion(s):
    code = s[-2:]

    if code == 'PM' and not s.startswith('12'):
        s = str(int(s[:2]) + 12) + s[2:]
    if s.startswith('12') and s.endswith('AM'):
        s = '00' + s[2:]
    return s[:-2]


if __name__ == '__main__':
    f = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = timeConversion(s)

    f.write(result + '\n')

    f.close()

File: ProblemSolving/Go/Warmup/time_conversion.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"strings"
	"strconv"
)

func timeConversion(s string) string {
	if strings.HasSuffix(s, "AM") {
		if strings.HasPrefix(s, "12") {
			return "00" + s[2:len(s)-2]
		}
		return s[:len(s)-2]
	} else {
		if strings.HasPrefix(s, "12") {
			return s[:len(s)-2]
		}
		v, _ := strconv.Atoi(s[:2])
		return strconv.Itoa(v+12) + s[2:len(s)-2]
	}
}

func main() {
	reader := bufio.NewReaderSize(os.Stdin, 16 * 1024 * 1024)

	stdout, err := os.Create(os.Getenv("OUTPUT_PATH"))
	checkError(err)

	defer stdout.Close()

	writer := bufio.NewWriterSize(stdout, 16 * 1024 * 1024)

	s := readLine(reader)

	result := timeConversion(s)

	fmt.Fprintf(writer, "%s\n", result)

	writer.Flush()
}

func readLine(reader *bufio.Reader) string {
	str, _, err := reader.ReadLine()
	if err == io.EOF {
		return ""
	}

	return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
	if err != nil {
		panic(err)
	}
}
